src/include/storage/storage_structure/disk_array.h

```diff
@@ -101,32 +101,29 @@ struct PIPUpdates {
  *   <li> apPageIdx: the physical disk pageIdx of some PIP.
  * </ul>
  */
-class BaseDiskArrayInternal {
+class DiskArrayInternal {
 public:
-    // Used by copiers.
-    BaseDiskArrayInternal(FileHandle& fileHandle, common::page_idx_t headerPageIdx,
-        uint64_t elementSize);
     // Used when loading from file
-    BaseDiskArrayInternal(FileHandle& fileHandle, DBFileID dbFileID,
-        common::page_idx_t headerPageIdx, BufferManager* bufferManager, WAL* wal,
-        transaction::Transaction* transaction, bool bypassWAL = false);
+    DiskArrayInternal(FileHandle& fileHandle, DBFileID dbFileID, common::page_idx_t headerPageIdx,
+        BufferManager* bufferManager, WAL* wal, transaction::Transaction* transaction,
+        bool bypassWAL = false);
 
-    virtual ~BaseDiskArrayInternal() = default;
+    virtual ~DiskArrayInternal() = default;
 
     uint64_t getNumElements(
         transaction::TransactionType trxType = transaction::TransactionType::READ_ONLY);
 
-    void get(uint64_t idx, transaction::TransactionType trxType, std::span<uint8_t> val);
+    void get(uint64_t idx, transaction::TransactionType trxType, std::span<std::byte> val);
 
     // Note: This function is to be used only by the WRITE trx.
-    void update(uint64_t idx, std::span<uint8_t> val);
+    void update(uint64_t idx, std::span<std::byte> val);
```

Comment:
Maybe. I'd included it just to see how well it would work (and also to see how it shows up in debuggers, which I never actually checked).
There isn't really anything it can do that `uint8_t` cannot, but it might be more appropriate since it's just meant to be used for representing raw memory; as cppreference says:
> unlike unsigned char, it is not a character type and is not an arithmetic type